arxiv-opensearch-cohere/README.md

```diff
@@ -0,0 +1,256 @@
+                "dimension": VECTOR_SIZE,
+                "method": {
+                    "name": "hnsw",
+                    "space_type": "l2",
+                    "engine": "nmslib",
+                    "parameters": {"ef_construction": 128, "m": 24},
+                },
+            },
+        }
+    },
+}
+response = client.indices.create(INDEX_NAME, body=body)
+```
+You can sanity check your index has been created by running the following line to get all indices currently populated in your OpenSearch backend. 
+```python
+print(client.indices.get_alias("*").keys())
+```
+
+Now that the index is created, we need to populate it with data. We are going to use the `cache.jsonl` we created in the previous step and the dataset file to populate the index with documents and their corresponding embedding vectors: 
+```python
+with open("cache.jsonl", "r") as fp:
+    cache = json.load(fp)
+
+# insert each row one-at-a-time to the document index
+for i, row in tqdm(df.iterrows()):
+    text = row.abstract
+    try:
+        embed = cache[text]
+        body = {
+            VECTOR_NAME: embed,
+            "text": text,
+            "title": row.title,
+            "arxiv_id": row.id,
+            "doi": row.doi,
+        }
+        response = client.index(index=INDEX_NAME, id=i, body=body)
+    except Exception as e:
+        print(f"[ERROR]: {e}")
+        continue
+```
+We have included the `abstract`, `title`, `arxiv_id` and `doi` fields in the document index from the data file. Feel free to include more or less depending on what you think you would want to search for. 
+
+Using the [opensearch-py-ml](https://pypi.org/project/opensearch-py-ml/) package, we are able to query our OpenSearch cluster with a pandas-like interface. You can run the following to ensure the documents inserted at your `index_name` are what you expect in terms of size. 
+
+**Note:** when using `opensearch-py-ml` package, ensure you look through the [documentation](https://opensearch-project.github.io/opensearch-py-ml/reference/index.html) as you cannot do every command as you would in pandas. 
+
+```python
+oml_df = oml.DataFrame(client, INDEX_NAME)
+print(oml_df.shape)
+```
+
+Your index has been created! 
+
+Full script can be run with `python create_index.py`. 
+
+## Step 4: Query your index for similar documents using cohere embeddings
+
+Now that your index is populated, you can proceed to querying it!
+Since we are using vectors for the kNN search, any of our queries must be translated to a vector using Cohere. The query vector is then submitted to the query endpoint within OpenSearch to find similar vectors via Approximate k-NN. 
+
+We can do so with the following helper functions: 
+```python
+
+def get_cohere_embedding(text: str) -> List[float]:
+    """
+    Embed a single text with cohere client and return list of float32
+    """
+    embed = np.array(co.embed([text], model=COHERE_MODEL).embeddings)
+    return list(embed.reshape(-1))
```

Comment:
can we simplify this to just be 
return co.embed([text], model=COHERE_MODEL).embeddings
i thought that just comes as a list?